### ISSUE TITLE
timeouts - same host smtp to mailserver

### ISSUE THREAD

**Issue by stephenmoloney** [tomav/docker-mailserver#498]
Not sure how/why this error comes about but an email is being sent by the django backend of a cabot app application that is running on the same host as `tomav/docker-mailserver`

```
Jan 29 18:55:44 mail postfix/smtp[1491]: connect to mail.example.com[host_ip_address]:25: Connection timed out
Jan 29 18:55:44 mail postfix/smtp[1491]: 4BC54182706: to=<amavis@mail.example.com>, orig_to=<amavis>, relay=none, delay=2259, delays=2229/0.02/30/0, dsn=4.4.1, status=deferred (connect to mail.example.com[[host_ip_address]:25: Connection timed out)
```

The python error is:

```
[2017-01-29 23:38:59,772: ERROR/Worker-13] Could not send Email alert: SoftTimeLimitExceeded()
Traceback (most recent call last):
  File "/code/cabot/cabotapp/alert.py", line 40, in send_alert
    alert.send_alert(service, users, duty_officers)
  File "/usr/local/lib/python2.7/site-packages/cabot_alert_email/models.py", line 49, in send_alert
    recipient_list=emails,
  File "/usr/local/lib/python2.7/site-packages/django/core/mail/__init__.py", line 50, in send_mail
    connection=connection).send()
  File "/usr/local/lib/python2.7/site-packages/django/core/mail/message.py", line 276, in send
    return self.get_connection(fail_silently).send_messages([self])
  File "/usr/local/lib/python2.7/site-packages/django/core/mail/backends/smtp.py", line 87, in send_messages
    new_conn_created = self.open()
  File "/usr/local/lib/python2.7/site-packages/django/core/mail/backends/smtp.py", line 48, in open
    local_hostname=DNS_NAME.get_fqdn())
  File "/usr/local/lib/python2.7/smtplib.py", line 256, in __init__
    (code, msg) = self.connect(host, port)
  File "/usr/local/lib/python2.7/smtplib.py", line 316, in connect
    self.sock = self._get_socket(host, port, self.timeout)
  File "/usr/local/lib/python2.7/smtplib.py", line 291, in _get_socket
    return socket.create_connection((host, port), timeout)
  File "/usr/local/lib/python2.7/socket.py", line 566, in create_connection
    sock.connect(sa)
  File "/usr/local/lib/python2.7/socket.py", line 228, in meth
    return getattr(self._sock,name)(*args)
  File "/usr/local/lib/python2.7/site-packages/billiard/pool.py", line 235, in soft_timeout_sighandler
    raise SoftTimeLimitExceeded()
SoftTimeLimitExceeded: SoftTimeLimitExceeded()
```

I am not a python guy.

**Comment by stephenmoloney** [comment]
I think this is related to or the same issue as #386.
I'm not sure of the solution though for #386 ?

**Comment by tomav** [comment]
I can be available this week end on Gitter of needed. Let me know.

**Comment by weaktyper** [comment]
I've done a bit more investigation on this issue. 

I can confirm that the issue is not really with `tomav/docker-mailserver` but I still feel like something is off. 

So I tested sending an email with python on the fly:

```
import smtplib
server = smtplib.SMTP("mail.<your_email_server>.com:587")
server.set_debuglevel(1)
server.ehlo()
server.starttls()
FROM = "username@.<your_email_server>.com"
TO = [email addresses]
server.login("username@<your_email_server>.com","PASSWORD")
server.sendmail(FROM, to, "email to self !!")
server.quit()
```
The above worked flawlessly. So really this library cannot be at fault.

So the problem I am having is that when I enter my email details into various applications (self-hosted applications), I cannot receive emails from those applications using the same settings. This is what is confusing me. My guess is that the clients are not doing the starttls step and perhaps this is not automatic. But literally all the clients do not take that step. Service providers such as `mailgun` do work however which leaves me more confused. I have blocked plain auth on port 25 by setting using the line `smtpd_tls_auth_only = yes`.

Any thoughts?

**Comment by stephenmoloney** [comment]
I've done a bit more investigation on this issue. 

I can confirm that the issue is not really with `tomav/docker-mailserver` but I still feel like something is off. 

So I tested sending an email with python on the fly:

```
import smtplib
server = smtplib.SMTP("mail.<your_email_server>.com:587")
server.set_debuglevel(1)
server.ehlo()
server.starttls()
FROM = "username@.<your_email_server>.com"
TO = [email addresses]
server.login("username@<your_email_server>.com","PASSWORD")
server.sendmail(FROM, to, "email to self !!")
server.quit()
```
The above worked flawlessly, the connection is made and an email is sent. Specifically the line `Feb 10 23:18:35 mail postfix/submission/smtpd[1522]` appears in the logs. So really this library cannot be at fault.

So the problem I am having is that when I enter my email details into various applications (self-hosted applications), I cannot receive emails from those applications using the client applications - various open source projects on github. This is what is confusing me. My guess is that the clients are not doing the starttls step and perhaps this is not automatic. But literally all the clients do not work. Service providers such as `mailgun` do work however which leaves me more confused. I have blocked plain auth on port 25 by setting using the line `smtpd_tls_auth_only = yes`. The line `Feb 10 23:18:35 mail postfix/submission/smtpd[1522]` never appears which I presume means there is a problem with these various clients connecting. The reason = ? (absence of steps to do STARTLS)? 

Any thoughts?
